api/src/main/scala/dcos/metronome/api/Authorization.scala

```diff
@@ -38,9 +38,11 @@ case class AuthorizedRequest[Body](identity: Identity, request: Request[Body], a
   def isAllowed(jobSpec: JobSpec): Boolean = authorizer.isAuthorized(identity, ViewRunSpec, jobSpec)
   def isAllowed(jobRun: JobRun): Boolean = isAllowed(jobRun.jobSpec)
   def isAllowed(started: StartedJobRun): Boolean = isAllowed(started.jobRun.jobSpec)
+  // QueuedJobRunInfo extends RunSpec so we can pass it directly to Authorizer
+  def isAllowed(queuedJobRunInfo: QueuedJobRunInfo): Boolean = authorizer.isAuthorized(identity, ViewRunSpec, queuedJobRunInfo)
 
   object Authorized {
-    def unapply(jobSpec: JobSpec): Option[JobSpec] = if (isAllowed(jobSpec)) Some(jobSpec) else None
+    def unapply(jobSpec: JobSpec): Option[JobSpec] = Some(jobSpec).filter(isAllowed)
```

Comment:
Is this extractor actually used somewhere? Don't see the usage in the diff but maybe it's somewhere else. If it is, never mind. Was just wondering.

Reply:
I just saw a refactoring suggestion by InteliJ on this line so I refactored it. So it was not directly related to this PR...